python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -488,6 +488,43 @@ def test_ndarray_nested_numpy_double(from_pandas, inner_seq):
                                 [[1., 2.], [1., 2., 3.], [np.nan], None])
 
 
+def test_nested_ndarray_in_object_array():
+    # ARROW-4350
+    arr = np.empty(2, dtype=object)
+    arr[:] = [np.array([1, 2]), np.array([2, 3])]
```

Comment:
Perhaps you should type this explicitly, e.g. `np.int64([1, 2])`. There is no guarantee that Numpy would choose `int64` on all platforms.

Reply:
Will fix. Windows indeed chooses int32